### PR TITLE
Spaceships

### PR DESCRIPTION
**Pull request by coyotte508**
https://github.com/donkeytech/gaia-engine/milestone/1

Spaceship expansion!

### CODE REVIEW

File: src/engine.ts [hunk 1]
```diff
@@ -1274,6 +1274,23 @@ export default class Engine {
     this.player(player).placeShip(hex);
   }
 
+  _move(player: PlayerEnum, ship: string, dest: string) {
```
Comment (rzulian) [inline src/engine.ts]: Why not a [Command.MoveShip]?

Comment (rzulian) [reply src/engine.ts]: :) Solved later

File: src/available-command.ts [hunk 2]
```diff
@@ -212,23 +214,40 @@ export function possibleShips(engine: Engine, player: Player) {
...
   }];
+
+  // Free action to boost ship
+  if (!pl.data.qicUsedToBoostShip && pl.canPay([new Reward("1q")])) {
+    commands.unshift({
+      name: Command.Spend,
+      player,
+      data: { acts: [{cost: "1q", income: "range+2"}] }
+    });
+  }
```
Comment (rzulian) [inline src/available-command.ts]: Wouldn't it better to move this under ```possibleFreeActions```?

**Comment by rzulian** [comment]
Awesome!!!!!!!
I cannot check the mechanism of this expansion, but it's definitely true that the overall architecture of the engine was flexible enough to be extend with this massive expansion!

File: src/engine.ts (hunk 1 above)
Comment (coyotte508) [reply src/engine.ts]: The reason is that there is already a Engine.move command, that would conflict with this command. Hence why I add to do something like [`_${Command.MoveShip]]`. We could also replace all the commands by [`command${Command.XXX}`] and phases by [`phase${Phase.XXX}`] to make it more generic!

**Pull request merged**